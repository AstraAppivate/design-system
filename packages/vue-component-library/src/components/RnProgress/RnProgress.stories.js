import { storiesOf } from '@storybook/vue'

import RnProgress from './index.vue'

storiesOf('Progress Bars', module)
  .add('default', () => ({
    components: { RnProgress },
    template: '<rn-progress value=25></rn-progress>',
  }))
  .add('small', () => ({
    components: { RnProgress },
    template: '<rn-progress value="50" size="small"></rn-progress>',
  }))
  .add('large', () => ({
    components: { RnProgress },
    template: '<rn-progress value="75" size="large"></rn-progress>',
  }))
