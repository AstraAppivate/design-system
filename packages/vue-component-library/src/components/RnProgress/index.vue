<template lang="html">
  <div class="rn-progress" :class="size">
    <div class="rn-progress__bar" :style="{ width: percentage }" />
  </div>
</template>

<script src="./scripts.js"></script>
