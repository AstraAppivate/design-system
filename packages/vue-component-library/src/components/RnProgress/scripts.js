export default {
  name: 'RnProgress',
  props: {
    value: {
      type: String,
      default: '0',
    },
    max: {
      type: String,
      default: '100',
    },
    min: {
      type: String,
      default: '0',
    },
    size: {
      type: String,
      default: 'default',
    },
  },
  computed: {
    percentage() {
      const max = parseInt(this.max, 10)
      const min = parseInt(this.min, 10)
      const value = parseInt(this.value, 10)

      const percent = (value / (max - min)) * 100
      return `${percent}%`
    },
  },
}
