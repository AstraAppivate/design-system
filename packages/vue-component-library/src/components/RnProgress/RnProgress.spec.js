import { shallow } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer'
import RnProgress from '@/src/components/RnProgress/index.vue'

describe('RnProgress.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(RnProgress, {
      propsData: {
        value: 50,
      },
    })
  })

  it.skip('snapshot: has same HTML structure', () => {
    const renderer = createRenderer()

    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) {
        throw new Error(err)
      }

      expect(str).toMatchSnapshot()
    })
  })
})
