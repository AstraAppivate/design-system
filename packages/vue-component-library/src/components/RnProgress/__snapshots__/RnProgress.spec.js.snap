// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RnProgress.vue snapshot: has same HTML structure 1`] = `
<div class="rn-progress default">
  <div class="rn-progress__bar" style="width:50%;"></div>
</div>
`;
